src/blueprints/Building.cpp

```diff
@@ -9,13 +9,22 @@
 bool Building::Build(Order* order_) {
     // Find place to build the structure
     sc2::Point3D base = gAPI->observer().StartingLocation();
-    sc2::Point2D point;
+    sc2::Point2D point, offset;
+    auto range = 15.0f; //greater than 15.0f may build too far away
+    auto buffer = 6.0f; //less than 6.0f may build in mineral line
 
     unsigned attempt = 0;
     do {
-        point.x = base.x + sc2::GetRandomScalar() * 15.0f;
-        point.y = base.y + sc2::GetRandomScalar() * 15.0f;
-
+        offset.x = sc2::GetRandomScalar() * range; //build in range of StartLoc
```

Comment:
I would suggest to try the approach used in CommandCenter.
The mineral line in the starting location is enclosed into a rectangle, if a building point is in the rectangle, different value is used. However, such approach requires careful detection of mineral lines which could be done in 
https://github.com/alkurbatov/suvorov-bot/blob/1fb4e7e36f538789bfd8b24f23b1c643637123c6/src/core/Map.h#L19

Reply:
Also we can use the placement grid abstraction from the api and mark the related tiles as occupied.

Reply:
I will look at that. I've reverted that commit, so the Reaper fix can be accepted while I work on this.